youtube_dl/extractor/pbs.py

```diff
@@ -466,7 +467,7 @@ def _extract_webpage(self, url):
                 url, display_id, note='Downloading player page',
                 errnote='Could not download player page')
             video_id = self._search_regex(
-                r'<div\s+id="video_([0-9]+)"', player_page, 'video ID')
+                [r'<div\s+id="video_([0-9]+)"', r'"id":\s*"([0-9]+)"'], player_page, 'video ID')
```

Comment:
I've already pointed out: relax regex. It must match arbitrary quotes and whitespace.

Reply:
I'm sorry but I can't change the new regex otherwise it will point to other things that are very similar but incorrect.

Reply:
Actually this approach is redundant since id can be extracted right from `webpage`.

Reply:
In rare cases it cannot be extracted directly from the webpage (like embed video)

Reply:
That's not an excuse for not adding another id patern for extraction from `webpage` that works fine with your test. Also add a test with "rare" case.

Reply:
Yes, it seems that it works due to this commit (https://github.com/rg3/youtube-dl/commit/d7be7053082055a001b788453d66131a62692b55) which was published after my pull request but merge before me. However my test is valid without the previous commit.

I also maintain that my id extraction can serve but if you insist I can also remove it.

Reply:
Again: https://www.pbs.org/wgbh/masterpiece/episodes/victoria-s2-e1/ can be processed **without** downloading `player_page` by looking for media id in `webpage`.
This regex is invalid because it matches several ids, it the order changes you'll end up downloading different video.